gcloud/storage/acl.py

```diff
@@ -345,78 +346,84 @@ def get_entities(self):
         self._ensure_loaded()
         return list(self.entities.values())
 
-    def reload(self):
-        """Reload the ACL data from Cloud Storage.
+    @property
+    def connection(self):
+        """Compute the connection for API requests for this ACL.
 
         This is a virtual method, expected to be implemented by subclasses.
 
-        :raises: :class:`NotImplementedError`
+        :raises: :class:`NotImplementedError` if ``_connection`` attribute
+                 is not set on the instance.
+
         """
+        # Allow override for testing
+        connection = getattr(self, '_connection', None)
+        if connection is not None:
+            return connection
         raise NotImplementedError
 
-    def save(self, acl=None):
-        """A method to be overridden by subclasses.
+    @property
+    def reload_path(self):
+        """Compute the path for GET API requests for this ACL.
 
-        :type acl: :class:`gcloud.storage.acl.ACL`, or a compatible list.
-        :param acl: The ACL object to save.  If left blank, this will save
-                    current entries.
+        This is a virtual method, expected to be implemented by subclasses.
 
-        :raises: NotImplementedError
+        :raises: :class:`NotImplementedError` if ``_reload_path`` attribute
+                 is not set on the instance.
         """
+        # Allow override for testing
+        path = getattr(self, '_reload_path', None)
+        if path is not None:
+            return path
         raise NotImplementedError
 
-    def clear(self):
-        """Remove all entities from the ACL."""
-        raise NotImplementedError
-
+    @property
+    def save_path(self):
+        """Compute the path for PATCH API requests for this ACL.
 
-class BucketACL(ACL):
-    """An ACL specifically for a bucket."""
+        This is a virtual method, expected to be implemented by subclasses.
 
-    _URL_PATH_ELEM = 'acl'
+        :raises: :class:`NotImplementedError` if ``_save_path`` attribute
+                 is not set on the instance.
 
-    def __init__(self, bucket):
         """
-        :type bucket: :class:`gcloud.storage.bucket.Bucket`
-        :param bucket: The bucket to which this ACL relates.
+        # Allow override for testing
+        path = getattr(self, '_save_path', None)
+        if path is not None:
+            return path
+        raise NotImplementedError
+
+    def reload(self, connection=None):
+        """Reload the ACL data from Cloud Storage.
+
+        :type connection: :class:`gcloud.storage.connection.Connection` or None
+        :param connection: explicit connection to use for API request;
+                           defaults to instance property.
         """
-        super(BucketACL, self).__init__()
-        self.bucket = bucket
+        if connection is None:
+            connection = self.connection
```

Comment:
Same comment here as I had in #852, this is not the same spirit as the changes in #843.